include/powsybl/iidm/ThreeWindingsTransformerLeg.hpp

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_THREEWINDINGSTRANSFORMERLEG_HPP
+#define POWSYBL_IIDM_THREEWINDINGSTRANSFORMERLEG_HPP
+
+#include <powsybl/iidm/CurrentLimitsAdder.hpp>
+#include <powsybl/iidm/PhaseTapChangerHolder.hpp>
+#include <powsybl/iidm/RatioTapChangerHolder.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+class Terminal;
+class ThreeWindingsTransformer;
+class ThreeWindingsTransformerAdder;
+
+namespace three_windings_transformer {
+
+class Leg : public virtual RatioTapChangerHolder, public virtual PhaseTapChangerHolder {
+public:  // Validable
+    std::string getMessageHeader() const override;
+
+public:  // TapChangerHolder
+    const Network& getNetwork() const override;
+
+    Network& getNetwork() override;
+
+public:  // RatioTapChangerHolder
+    const RatioTapChanger& getRatioTapChanger() const override;
+
+    RatioTapChanger& getRatioTapChanger() override;
+
+    bool hasRatioTapChanger() const override;
+
+    RatioTapChangerAdder newRatioTapChanger() override;
+
+public:  // PhaseTapChanger
+    const PhaseTapChanger& getPhaseTapChanger() const override;
+
+    PhaseTapChanger& getPhaseTapChanger() override;
+
+    bool hasPhaseTapChanger() const override;
+
+    PhaseTapChangerAdder newPhaseTapChanger() override;
+
+public:
+    Leg(unsigned long legNumber, double r, double x, double g, double b, double ratedU, double ratedS);
+
+    Leg(const Leg&) = delete;
+
+    Leg(Leg&&) noexcept = default;
```

Comment:
To check: do we have the same issue and need to update the backward reference. Under linux, it seems the object is reused, but I'm not sure about the behaviour under windows

Reply:
This is not a problem unlike for TieLineAdder because the Leg is a validable and directly emits errors by passing `*this`.
It would be a problem if the 3WT would be moved (transformer reference should be updated) but it is not specific to those classes (3WT / Leg) => https://github.com/powsybl/powsybl-iidm4cpp/issues/235